handlers/k8sconfig_handler.go

```diff
@@ -293,41 +301,69 @@ func (h *Handler) GetCurrentContext(token string, prov models.Provider) (*models
 				logrus.Warn("failed to save the context: ", err)
 				continue
 			}
+			if !viper.GetBool("SKIP_COMP_GEN") {
+				k8ctxID := ctx.ID
+				cfg, err := ctx.GenerateKubeConfig()
+				if err != nil {
+					return &ctx, nil
+				}
 
-			if ctx.IsCurrentContext {
-				_, _ = prov.SetCurrentContext(token, ctx.ID) // Ignore the error
-
-				if !viper.GetBool("SKIP_COMP_GEN") {
-					ctxID := ctx.ID
-					cfg, err := ctx.GenerateKubeConfig()
+				go func(config []byte, ctx string) {
+					compCreationSingleton.cancelPreviousRun(ctx) //if a registerK8sComponents is still running for the same context then we safely cancel that assuming that run to be stale now
+					ctxt, cancel := context.WithTimeout(context.Background(), 600*time.Second)
+					compCreationSingleton.registerCancelFunc(ctx, cancel)
+					h.log.Info("Starting to register k8s native components for contextID ", ctx)
+					err := registerK8sComponents(ctxt, config, k8ctxID)
 					if err != nil {
-						return &ctx, nil
+						h.log.Error(err)
+						return
 					}
-
-					go func(l logger.Handler, config []byte, ctx string) {
-						err := registerK8sComponents(h.log, config, ctxID)
-						if err != nil {
-							logrus.Error(err)
-						}
-					}(h.log, cfg, ctxID)
-				}
-
-				return &ctx, nil
+					h.log.Info("Registration of k8s native components completed for contextID ", ctx)
```

Comment:
Are we ensuring that these are Kubernetes native and not custom resources?

Reply:
Yes

Reply:
We cannot do that. All api-resources are treated similarly by kubernetes whether those are custom or k8s native.
When kubernetes cannot distinguish, so cant we. And we should not as well, although more importantly we cannot.

Reply:
We can and will.